x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRCS1PainlessExecuteIT.java

```diff
@@ -0,0 +1,276 @@
+            final var putUserRequest = new Request("PUT", "/_security/user/" + REMOTE_SEARCH_USER);
+            putUserRequest.setJsonEntity("""
+                {
+                  "password": "x-pack-test-password",
+                  "roles" : ["remote_search"]
+                }""");
+            assertOK(adminClient().performRequest(putUserRequest));
+
+            // Query cluster -> create test index
+            final var indexDocRequest = new Request("POST", "/local_index/_doc?refresh=true");
+            indexDocRequest.setJsonEntity("{\"local_foo\": \"local_bar\"}");
+            assertOK(client().performRequest(indexDocRequest));
+
+            // Fulfilling cluster -> create test indices
+            final Request bulkRequest = new Request("POST", "/_bulk?refresh=true");
+            bulkRequest.setJsonEntity(Strings.format("""
+                { "index": { "_index": "index1" } }
+                { "foo": "bar" }
+                { "index": { "_index": "secretindex" } }
+                { "bar": "foo" }
+                """));
+            assertOK(performRequestAgainstFulfillingCluster(bulkRequest));
+        }
+
+        {
+            // TEST CASE 1: Query local cluster for local_index - should work since role has read perms for it
+            Request painlessExecuteLocal = createPainlessExecuteRequest("local_index");
+
+            Response response = performRequestWithRemoteSearchUser(painlessExecuteLocal);
+            assertOK(response);
+            String responseBody = EntityUtils.toString(response.getEntity());
+            assertThat(responseBody, equalTo("{\"result\":[\"test\"]}"));
+        }
+        {
+            // // TEST CASE 2: Query remote cluster for index1 - should fail since no permissions granted for remote clusters yet
+            // Request painlessExecuteRemote = createPainlessExecuteRequest("my_remote_cluster:index1");
+            // Exception exc = expectThrows(Exception.class, () -> performRequestWithRemoteSearchUser(painlessExecuteRemote));
+            // assertThat(exc, notNullValue());
+            // assertThat(exc.getMessage(), containsString("Connection is closed")); // TODO: is this the expected exception?
+
+            // assertThat(exc.getResponse().getStatusLine().getStatusCode(), is(403));
+            // String errorResponseBody = EntityUtils.toString(exc.getResponse().getEntity());
+            // assertThat(err orResponseBody, containsString("unauthorized for user [remote_search_user]"));
+            // assertThat(errorResponseBody, containsString("on indices [secretindex]"));
+            // assertThat(errorResponseBody, containsString("\"type\":\"security_exception\""));
+            //
+            // Response response = performRequestWithRemoteSearchUser(painlessExecuteRemote);
+            // String responseBody = EntityUtils.toString(response.getEntity());
+            // assertOK(response);
+            // assertThat(responseBody, equalTo("{\"result\":[\"test\"]}"));
```

Comment:
Let's decide what to do with these test cases before merging, rather than just leaving them commented out like this. If we care about them, let's get them working (or at least open an issue to report the problem they represent). If we don't care about them, let's just remove them.

Reply:
Yes, I'm still working on the tests with @slobodanadamovic

Reply:
Tests are updated now.